app/src/androidTest/java/com/karumi/katasuperheroes/MainActivityTest.java

```diff
@@ -65,6 +85,70 @@
     onView(withText("¯\\_(ツ)_/¯")).check(matches(isDisplayed()));
   }
 
+  @Test public void showsEmptyCaseWhenThereAreSuperHeroes(){
+    givenThereAreOneSuperHeroe();
+    startActivity();
+    onView(withText("¯\\_(ツ)_/¯")).check(matches(not(isDisplayed())));
+  }
+
+  @Test public void viewShowsSuperHeroeName(){
+    givenThereAreOneSuperHeroe();
+    startActivity();
+    onView(withId(R.id.tv_super_hero_name)).check(matches(isDisplayed()));
+  }
+
+  @Test public void showSuperHeroesNameIfThereAreSuperHeroes(){
+    final List<SuperHero> superHeroes = giverNSuperHeroes(100);
```

Comment:
If the number of super heroes is not important for this test you can use a default value hidden to the ``givenNSuperHeres`` method user.